### PR TITLE
JPA storage can return list of all artifacts

### PR DESCRIPTION
**Pull request by dang-gyg**
the method `getArtifactIds()` is there, but it is not defined in the `openapi.json`. This PR adds the GET endpoint to return all published artifacts, and links it to the existing JPA implementation

### CODE REVIEW

**Comment by dang-gyg** [comment]
@alesj @EricWittmann I forgot to add reviewers. Can you help me add, please? thank you.

**Comment by EricWittmann** [comment]
We actually left out the `getArtifacts` call from the API intentionally.  The idea was to add a search API later that would drive e.g. a user interface.  However, I could be convinced that it makes sense to have this simple list of artifact IDs (the confluent API has this).  What do you think @alesj and @jsenko ?

**Comment by alesj** [comment]
I’m good with whatever you think is best 🙂

File: client/src/main/java/io/apicurio/registry/client/CachedRegistryService.java [hunk 1]
```diff
@@ -139,6 +135,11 @@ public VersionMetaData getArtifactVersionMetaData(Integer version, String artifa
         });
     }
 
+    @Override
+    public Set<String> getArtifactIds() {
+        return getDelegate().getArtifactIds();
```
Comment (alesj) [inline client/src/main/java/io/apicurio/registry/client/CachedRegistryService.java]: Perhaps we could cache this ids as well ...

**Comment by EricWittmann** [comment]
Yeah I agree with that concern - and I think it's why we didn't want to include the functionality originally.  But since the confluent compatibility API already has this capability (with no support for paging afaik) I'm not sure it's necessary.

Perhaps in both cases we put in a sensible (configurable?) LIMIT to avoid insanity.  Something like 1000 perhaps?  That shouldn't take very long to produce, since the response payload is just an array of IDs.  But we do want to avoid this call taking a long time if there are e.g. 10 million schemas in the registry.  :)

**Comment by alesj** [comment]
That's why I had that TODO in the code ... "filter" ... :-)
Perhaps along with LIMIT, we could also add optional regexp for the name?

**Comment by alesj** [comment]
Although, distributed storages, such as Streams, don't handle LIMIT well, as that's not really deterministic ... or at least in those cases hard to achieve.

**Comment by EricWittmann** [comment]
Yeah a limit on the in-memory storage impls will likely be pretty non-deterministic.  But I think the idea is just to have a sanity check.  So once the number of artifacts grows beyond the limit, what you get back may not be very helpful, and it may be a different set of stuff each time you ask.  But I would think that the use-case for a non-paged query returning all of the artifact IDs is pretty limited anyway.

A better way to get a list of artifacts is coming once the search/UI stuff is done.  At that point we can point to that API for use-cases here filtering and paging are needed.

**Comment by EricWittmann** [comment]
My suggestion is that we do this:

* Merge this as-is
* Add a GitHub issue to implement "limit"
* Keep making progress on UI and search

Thoughts?

**Comment by alesj** [comment]
Sounds good

**Comment by dang-gyg** [comment]
thanks for the feedback. can you share more detail on the UI feature? is it gonna similar to Redoc?

**Comment by EricWittmann** [comment]
Actually I imagine we'll probably embed ReDoc (for OpenAPI docs) as one of the visualizers.  But we'll need other solutions to visualize the other supported formats.

Basically the UI will be a listing of all the artifacts (with searching, paging, filtering, etc) and then a detail page of some sort per-artifact.  The detail page will have all the meta-data about the artifact in addition to a way to see generated docs (e.g. ReDoc) and perform some actions (download, generate client sdk, etc).

**Comment by dang-gyg** [comment]
cool. I look forward to the official support. In the meantime I quipped up ReDoc support for OpenAPI specs by implementing `GET /docs/{artifactId}/` and applying the template here https://github.com/Redocly/redoc/blob/master/config/docker/index.tpl.html. It is working fairly well :)

**Comment by EricWittmann** [comment]
As discussed:  https://github.com/Apicurio/apicurio-registry/issues/328

**Pull request merged**